### PR TITLE
feat(PX-4135): Collector can set an address as default

### PR DESCRIPTION
**Pull request by Serge0n**
The collector can set an address as default

Jira: https://artsyproduct.atlassian.net/browse/PX-4135

<details>
  <summary>Edit modal when isDefault true</summary>

https://user-images.githubusercontent.com/55637696/125309131-2b5c2700-e33a-11eb-93ea-8fb0e158f126.mov
</details>

<details>
  <summary>Edit modal when isDefault false</summary>

https://user-images.githubusercontent.com/55637696/125309297-50e93080-e33a-11eb-891e-920d9408d39d.mov
</details>

<details>
  <summary>Add modal(checkbox is always shown)</summary>

https://user-images.githubusercontent.com/55637696/125309822-ba693f00-e33a-11eb-98c4-0654d9a12cb7.mov
</details>

<details>
  <summary>Correct state when uncheck</summary>

https://user-images.githubusercontent.com/55637696/125309372-6199a680-e33a-11eb-86f5-16da6a34f9a4.mov
</details>

<details>
  <summary>How hover should work</summary>

https://user-images.githubusercontent.com/55637696/125445341-a91c8858-b8ea-45d7-8169-fb51ade8b8b4.mov
</details>

### CODE REVIEW

File: src/v2/Apps/Order/Components/AddressModal.tsx
Comment (pepopowitz) [inline src/v2/Apps/Order/Components/AddressModal.tsx:110]: blocking: this line was added in a bunch of places when we initially turned on strict null checking, but we should not be adding new ones. If this line is needed to prevent a linting error in this call, we should fix the arguments instead.

File: src/v2/Apps/Order/Components/AddressModal.tsx
Comment (pepopowitz) [inline src/v2/Apps/Order/Components/AddressModal.tsx:108]: Blocking if my understanding is correct: Do we not have access to the value through `values.isDefault`? It looks like we should ([here's the underlying type](https://github.com/artsy/force/blob/1db17a9b7ee6ff5c44401e0b46160d04cde68ab0/src/v2/__generated__/Shipping_me.graphql.ts#L20)). If we do, we should use that instead of creating another state variable to manage this value.

File: src/v2/Apps/Order/Components/SavedAddressItem.tsx
Comment (pepopowitz) [inline src/v2/Apps/Order/Components/SavedAddressItem.tsx:57]: Not blocking but pretty important visually: In the video named "Edit modal when isDefault false", the addresses jump around a bit when the pill moves from the 2nd to the 1st -- this makes me think there's a vertical margin or maxHeight or something else that's not quite right, or maybe even some missing margin on line 1 of the address text. Could we adjust this?

File: src/v2/Apps/Order/Components/__tests__/AddressModal.jest.tsx
Comment (pepopowitz) [inline src/v2/Apps/Order/Components/__tests__/AddressModal.jest.tsx:67]:
```suggestion
  it("renders EditModal without checkbox when address is default", () => {
```

(very minor)

File: src/v2/Apps/Order/Components/__tests__/AddressModal.jest.tsx
Comment (pepopowitz) [inline src/v2/Apps/Order/Components/__tests__/AddressModal.jest.tsx:80]:
```suggestion
    expect(wrapper.find("Checkbox[data-test='setAsDefault']").length).toBe(0)
```

Not blocking but fairly important: The majority of these assertions are already covered in other tests. The only one I see that is important here is to verify the checkbox isn't appearing in these conditions. 

Having more assertions here than necessary will lead to the test requiring changes for unrelated reasons.

**Review by pepopowitz — CHANGES_REQUESTED** [review]
Looks good! I pointed out a couple small things we should change.

File: src/v2/Apps/Order/Components/SavedAddressItem.tsx
Comment (damassi) [inline src/v2/Apps/Order/Components/SavedAddressItem.tsx:102]: Blocking: As this is a component in our design system the more constrained the better, meaning we should avoid styling palette components in this way. If we need to add the ability to control background colors we should update the palette `Pill` component directly. But more ideally, we push back at design on these one-off changes and instead go with current defaults.

File: src/v2/Apps/Order/Components/AddressModal.tsx
Comment (Serge0n) [reply src/v2/Apps/Order/Components/AddressModal.tsx:110]: fixed

File: src/v2/Apps/Order/Components/SavedAddressItem.tsx
Comment (Serge0n) [reply src/v2/Apps/Order/Components/SavedAddressItem.tsx:57]: I took the height value from Figma. I think 21 will be correct.

File: src/v2/Apps/Order/Components/SavedAddressItem.tsx
Comment (Serge0n) [reply src/v2/Apps/Order/Components/SavedAddressItem.tsx:102]: Yes, totally agree with this.  What should I do at the moment? Leave it to work like this? As I understand I can't make corrections for the Pill component at Palette by myself.

https://user-images.githubusercontent.com/55637696/125597431-cb2abeb5-86e8-4d2d-aefd-956c5bf2583d.mov

File: src/v2/Apps/Order/Components/AddressModal.tsx
Comment (Serge0n) [reply src/v2/Apps/Order/Components/AddressModal.tsx]: fixed

**Review by pepopowitz — APPROVED** [review]

File: src/v2/Apps/Order/Components/SavedAddressItem.tsx
Comment (damassi) [reply src/v2/Apps/Order/Components/SavedAddressItem.tsx:110]: I would suggest shipping this for now, and then we can iterate on adding a new property to Palette! Thoughts @pepopowitz?  

It would be worth raising in the #design-system channel in our slack so that designers (and other devs) can comment.

Comment (pepopowitz) [reply src/v2/Apps/Order/Components/SavedAddressItem.tsx:110]: A few thoughts: 

- +1 to shipping & iterating.
- Is there a bug here somewhere? In the video (1) the Pill starts with a medium gray background; (2) it gets a very light gray background when we hover over the second address box; (3) it stays light gray when you hover over the Pill directly; (4) it changes to a medium gray background when leaving the Pill but still hovering the address box. I'd expect 4 & 2 to match states, but they don't. If there's a bug within this PR that's causing it to turn light gray in step 2, I don't think the video would look as bad. (Is a hover state being propagated to the Pill somehow?) On the other hand, if the bug is within palette, we should track it there but I wouldn't expect to hold up this PR for it.
- It seems especially important to call out this exact demo to the design system group. The reason the default `Pill` styling doesn't look great is the combination of the Pill with an area that's turning gray due to hover. I'd expect the group to come back with either (1) we need an alternate style for the Pill because we'll be doing this more, or (2) this current UI could be re-worked to better utilize the components & design we already have, avoiding the need for an alternate `Pill` style. 
- It also would be worth raising to @tinn, to collaboratively work toward a short-term and long-term solution that utilizes palette completely instead of dropping down into styled-components to override the background. 

That's a lot, so to be clear, I wouldn't personally hold up a merge of either the styled-components solution or the solution in this video, as long as we track the work to follow up and make it better.

Comment (damassi) [reply src/v2/Apps/Order/Components/SavedAddressItem.tsx:110]: For reference, here's a link to [Palette's Pill storybook](https://palette-storybook.artsy.net/?path=/story/components-pill--states)

> as long as we track the work to follow up and make it better

I'd still encourage us to avoid merging the overrides with `!important` for the time being, unless there's a ticket and an immediate follow-up. The visual hover state is weird but it's noticeable, where stray overrides and `!important` could cause real confusion down the road, as they tend to do.

Comment (sweir27) [reply src/v2/Apps/Order/Components/SavedAddressItem.tsx:110]: I'm going to take a stab at writing up a ticket for this and pull it into the sprint as a follow up! Will ping you @pepopowitz to make sure it's accurate. 🙏

**Pull request merged**
